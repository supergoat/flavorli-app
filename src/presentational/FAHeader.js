// @flow
import React, { Component } from 'react';
import {
  SafeAreaView,
  Text,
  TouchableOpacity,
  View
} from 'react-native';
import FAColors from "../common/FAColors";
import FAStyleSheet from "../common/FAStyleSheet";

/** ============================================================================
<FAHeader />
--------------------------------------------------------------------------------
View Header
============================================================================= */
type Props = {}
class FAHeader extends Component<Props> {
  render() {
    return (
      <SafeAreaView style={styles.safeAreaView}>
        <View style={styles.header}>
          <Text style={styles.title}>flavorli</Text>
        </View>
      </SafeAreaView>
    )
  }
}

/* StyleSheet
============================================================================= */
const styles = FAStyleSheet.create({
  safeAreaView: {
    backgroundColor: FAColors.alabaster
  },
  header: {
    flexDirection: 'row',
    zIndex: 1, // Removes shadow that is sometimes caused by react-navigation
    backgroundColor: FAColors.alabaster,
    borderBottomWidth: 1,
    borderBottomColor: FAColors.gallery,
    paddingBottom: 5
  },
  title: {
    flex: 1,
    textAlign: 'center',
    color: FAColors.oxfordBlue,
    fontFamily: 'Pacifico-Regular',
    fontSize: 30,
  }
});

/* Export
============================================================================= */
export default FAHeader;
