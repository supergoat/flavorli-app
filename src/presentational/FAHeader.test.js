// @flow
import React from 'react';
import { View, Text } from 'react-native';
import FAHeader from './FAHeader';
import renderer from 'react-test-renderer';

describe('FAHeader', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<FAHeader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
