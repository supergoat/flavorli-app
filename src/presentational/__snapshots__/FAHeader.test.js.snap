// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAHeader renders correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "#F9F9F9",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#F9F9F9",
        "borderBottomColor": "#EEEEEE",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "paddingBottom": 5,
        "zIndex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#333C4E",
          "flex": 1,
          "fontFamily": "Pacifico-Regular",
          "fontSize": 30,
          "textAlign": "center",
        }
      }
    >
      flavorli
    </Text>
  </View>
</RCTSafeAreaView>
`;
